internal/subroutine/remove_initializer.go

```diff
@@ -48,6 +57,24 @@ func (r *removeInitializer) Process(ctx context.Context, instance runtimeobject.
 		return ctrl.Result{}, nil
 	}
 
+	// we need to wait until keycloak crossplane provider creates a portal secret
+	workspaceName := getWorkspaceName(lc)
+	if workspaceName == "" {
+		return ctrl.Result{}, errors.NewOperatorError(fmt.Errorf("failed to get workspace path"), true, false)
+	}
+
+	secretName := fmt.Sprintf("portal-client-secret-%s", workspaceName)
+	key := types.NamespacedName{Name: secretName, Namespace: PortalClientSecretNamespace}
+
+	var secret corev1.Secret
+	if err := r.runtimeClient.Get(ctx, key, &secret); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info().Msg(fmt.Sprintf("portal secret %s is not ready yet, trying again", secretName))
+			return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```

Comment:
I also think it would be good to transition to return an error after a certain time has passed, that would also be the time where a sentry notification should be send and a error log should be printed. Maybe we start with a 1 minute (if more then a minute has passed since logicalcluster creation)